### PR TITLE
[MRG] Write categories.xml

### PR DESCRIPTION
**Pull request by jusjusjus**
Hey Evan, I'm a bit unclear on a few points still; I'll refer you to the test to see how to use the code.

This PR addresses Issue #27

### CODE REVIEW

File: mffpy/tests/test_xml_files.py [hunk 1]
```diff
@@ -397,6 +397,59 @@ def test_sort_categories_by_starttime(categories, idx, expected):
...
+                        'data': 3
+                    },
+                    'subj': {
+                        'type': 'person',
+                        'data': 'RM271_noise_test'
+                    }
+                }
+            }
+        ],
+    }
```
Comment (jusjusjus) [inline mffpy/tests/test_xml_files.py:429]: This is the (clumsy) input to writing a "categories.xml".  My problem is that I don't fully understand what's included here, and which of the keys and items are unique.  Please check this input.

File: mffpy/xml_files.py [hunk 2]
```diff
@@ -834,6 +834,10 @@ class Categories(XML):
     _xmlns = r'{http://www.egi.com/categories_mff}'
     _xmlroottag = r'categories'
     _default_filename = 'categories.xml'
+    _type_converter = {
+        'long': int,
+        'person': str,
+    }
```
Comment (jusjusjus) [inline mffpy/xml_files.py:840]: Can you think of more?

File: mffpy/xml_files.py [hunk 3]
```diff
@@ -930,6 +947,91 @@ def sort_categories_by_starttime(self) -> List[dict]:
         sorted_categories.sort(key=lambda b: b['t0'])
         return sorted_categories
 
+    @classmethod
+    def content(cls, categories):
+
+        def serialize_segment(segment):
```
Comment (ephathaway) [inline mffpy/xml_files.py:953]: Can this be a private function?

File: mffpy/xml_files.py [hunk 4]
```diff
@@ -930,6 +947,91 @@ def sort_categories_by_starttime(self) -> List[dict]:
...
+                    'data': data
+                } for keyCode, data in keys_by_code.items()]
+                # convert xml element list
+                key_list = [{TEXT: item} for item in key_list]
+                # add to output
+                text['keys'] = {TEXT: {'key': key_list}}
+
+            return output
+
+        def serialize_category(args):
```
Comment (ephathaway) [inline mffpy/xml_files.py:1015]: Can this also be a private function?

File: mffpy/xml_files.py (hunk 2 above)
Comment (ephathaway) [reply mffpy/xml_files.py]: Looks like we can have type 'fileReference'
```
<key>
                        <keyCode>FILE</keyCode>
                        <data dataType="fileReference">RM271_noise_test_20190501_105754_seg.mff</data>
                    </key>
```

**Review by ephathaway — CHANGES_REQUESTED** [review]
Have a look at my PR #42 and see if you agree. I tested writing a categories.xml file, plugging it into an MFF, and opening it up in Net Station. It seems to not be breaking anything, so that is a good sign. Also passes EGI's file validator tool (but not sure how much that tells us).

**Comment by jusjusjus** [comment]
Hey @ephathaway , I made some changes to your suggestions, pulled these in, and addressed your other comments.

**Review by ephathaway — APPROVED** [review]
I think this is ready to go. I am able to parse a `categories.xml` file, write the content to a new MFF, parse the `categories.xml` from the new MFF and get the same result. I tried this on a segmented and averaged MFF file.

**Pull request merged**